libse/Subtitle.cs

```diff
@@ -580,5 +580,21 @@ public string GetFastHashCode()
             return sb.ToString().TrimEnd();
         }
 
+        public string GetAllText(bool appendNewLine)
+        {
+            var sb = new StringBuilder();
+            foreach (Paragraph p in Paragraphs)
+            {
+                if (appendNewLine)
+                {
+                    sb.AppendLine(p.Text);
+                }
+                else
+                {
+                    sb.Append(p.Text.TrimEnd() + ' ');

```

Comment:
I would probably go with:
```
        public string GetAllText(bool joinLines = false)
        {
            var sb = new StringBuilder();
            foreach (Paragraph p in Paragraphs)
            {
                if (joinLines)
                {
                    sb.Append(p.Text.TrimEnd()).Append(' ');
                }
                else
                {
                    sb.AppendLine(p.Text);
                }
            }
            return joinLines ? sb.ToString().TrimEnd() : sb.ToString();
        }
```
Or, if joining lines is not necessary, use a property:
```
        public string AllText
        {
            get
            {
                var sb = new StringBuilder();
                foreach (Paragraph p in Paragraphs)
                {
                    sb.AppendLine(p.Text);
                }
                return sb.ToString();
            }
        }
```
Of course, joined lines **could** even be a separate property:
```
        public string AllTextJoined
        {
            get
            {
                var sb = new StringBuilder();
                foreach (Paragraph p in Paragraphs)
                {
                    sb.Append(p.Text.TrimEnd()).Append(' ');
                }
                return sb.ToString().TrimEnd();
            }
        }
```
However, the question here is “What does Nikolaj prefer?” :wink: